ClientApp/viking-vault/src/display_user_list/AddMoneyModal.tsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import './AddMoney.css';
+
+interface IAddMoneyModalProps{
+    open: boolean;
+    userName: string;
+    closeModal: any;
+    addMoney: any;
+}
+
+interface IAddMoneyModalState{
+    amount: string;
+}
+
+class AddMoneyModal extends React.Component<IAddMoneyModalProps, IAddMoneyModalState>{ 
+    state = {
+            amount : ""
+        };
+
+    private resetAmountField(){
+        this.setState({
+            amount : ""
+        });
+    }
+    private closeModal = () =>{
+        this.resetAmountField();
+        this.props.closeModal(false);
+    }
+
+    private handleAddMoney = () =>{
+        this.resetAmountField();
+        this.props.addMoney(Number(this.state.amount));
+    }
+
+    private handleAmountChange = (e:any) =>{
+        this.setState({
+            amount: e.target.value
+        });
+    }
+
+    private validateAmount(){
+        if(Number(this.state.amount) <= 0 || Number(this.state.amount) > 5000 || !(this.state.amount.match("^[-+]?[0-9]*\.?[0-9]+([eE][-+]?[0-9]+)?$"))){
```

Comment:
Lower down it says that the minimum amount is 0.1 but here you only check for positive amount